app/src/main/java/com/kickstarter/viewmodels/usecases/GetShippingRulesUseCase.kt

```diff
@@ -233,10 +241,23 @@ class GetShippingRulesUseCase(
     ): ShippingRule =
         if (project.isBacking()) ShippingRule.builder()
             .apply {
+                val backing = project.backing()
                 val locationId = project.backing()?.locationId() ?: 0L
-                val locationName = project.backing()?.locationName() ?: ""
-
-                this.location(Location.Builder().id(locationId).name(locationName).displayableName(locationName).build())
+                this.id(locationId)
+                val reward = backing?.reward()?.let {
+                    if (RewardUtils.shipsToRestrictedLocations(it)) {
+                        val rule = backing?.reward()?.shippingRules()
+                            ?.first { it.location()?.id() == locationId }
+                        this.location(rule?.location())
+                        this.id(rule?.id())
+                        this.cost(rule?.cost() ?: 0.0)
+                    }
+                    if (RewardUtils.shipsWorldwide(it)) {
+                        val locationName = project.backing()?.locationName() ?: ""
```

Comment:
for shipping everywhere reward will have 2 shipping rules with location, name "Earth" an "Everywhere in the world", so take the location name and cost from backing.location, backing.shipping